app/src/main/java/com/example/henzoshimada/feeltrip/ElasticSearchController.java

```diff
@@ -32,18 +33,30 @@
     private static final String typeMood = "mood";
     private static final String typeUser = "user";
 
-    public static class AddMoodTask extends AsyncTask<Mood, Void, Void> {
+    public static class AddMoodTask extends AsyncTask<Mood, Void, Boolean> {
+
+        private Context context;
+
+        public AddMoodTask(Context context) {
+            this.context = context.getApplicationContext();
+        }
 
         @Override
-        protected Void doInBackground(Mood ... moods ) {
+        protected void onPostExecute(Boolean result) {
+            FeelTripApplication.loadFromElasticSearch();
+            FeelTripApplication.getMoodAdapter(context).notifyDataSetChanged();
+        }
```

Comment:
This block allows us to refresh the adapter once we're sure that elasticsearch has uploaded the current package. This fixes the bug where we post a mood, and don't immediately see the change of it on the homescreen.